toto/runlib.py

```diff
@@ -35,53 +35,63 @@
 else:
   import subprocess
 
-# XXX LP: I think we'll get rid of toto formats because we want to use
-# model validators instead for toto schemas, we'll still use
-# toto.ssl_crypto.formats though
-
-# import toto.formats
-
 import toto.models.link
 import toto.log as log
 
 import toto.ssl_crypto.hash
 import toto.ssl_crypto.formats
 
+def _hash_artifact(filepath, hash_algorithms=['sha256']):
+  """Internal helper that takes a filename and hashes the respective file's
+  contents using the passed hash_algorithms and returns a hashdict conformant
+  with ssl_crypto.formats.HASHDICT_SCHEMA. """
+  toto.ssl_crypto.formats.HASHALGORITHMS_SCHEMA.check_match(hash_algorithms)
+  hash_dict = {}
 
-def record_artifacts_as_dict(artifacts):
-  """Takes a list of directories and/or filenames as input and creates a dict
-  with filepaths as keys and their files' hashes as values.
+  for algorithm in hash_algorithms:
+    digest_object = toto.ssl_crypto.hash.digest_filename(filepath, algorithm)
+    hash_dict.update({algorithm: digest_object.hexdigest()})
 
-  The dirs/files a link command is executed on are called materials.
-  The dirs/files that result form a link command execution are called
-  products."""
+  toto.ssl_crypto.formats.HASHDICT_SCHEMA.check_match(hash_dict)
 
-  artifacts_dict = {}
+  return hash_dict
 
-  if not artifacts:
-    return artifacts_dict
 
-  def _hash_artifact(filepath, hash_algorithms=['sha256']):
-    """Takes a filename and hashes the respective file's contents
-    using the passed hash_algorithms. Returns a HASHDICT"""
+def _normalize_path(path):
+  """Internal helper that strips "./" on the left side of the path. """
+  if path.startswith("./"):
```

Comment:
Good point. I actually don't know how `os.walk(".")` works on windows, but on *NIX it prepends `./`.
There is already a ticket for this issue:
https://github.com/in-toto/in-toto/issues/15